internal/webhook/external/github_test.go

```diff
@@ -0,0 +1,134 @@
+package external
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/indexer"
+)
+
+func TestGithubHandler(t *testing.T) {
+	scheme := runtime.NewScheme()
+	require.NoError(t, kargoapi.AddToScheme(scheme))
+	kubeClient := fake.NewClientBuilder().
+		WithScheme(scheme).
+		WithObjects(
+		// not adding any objects
+		// because the refresh is tested in helpers_test.go
+		// this test just ensures the correct http status
+		// codes are returned for the edgecases provided
+		).
+		WithIndex(
+			&kargoapi.Warehouse{},
+			indexer.WarehousesBySubscribedURLsField,
+			indexer.WarehousesBySubscribedURLs,
+		).Build()
+	serverURL := "http://doesntmatter.com"
+
+	for _, test := range []struct {
+		name         string
+		setup        func() *http.Request
+		expectedCode int
+		expectedMsg  string
+	}{
+		{
+			name: "OK",
```

Comment:
Nit: When you look at test cases and the function under test side-by-side, it's much easier to follow the test cases when they're laid out in the same order as all the early returns from the function that they exercise.